modoboa_webmail/lib/imapheader.py

```diff
@@ -83,18 +84,22 @@ def parse_reply_to(value, **kwargs):
     """
     return parse_address_list(value, **kwargs)
 
-
 def parse_date(value, **kwargs):
     """Parse a Date: header."""
     tmp = email.utils.parsedate_tz(value)
     if not tmp:
         return value
-    tz = timezone.get_current_timezone()
-    ndate = tz.localize(
-        datetime.datetime.fromtimestamp(email.utils.mktime_tz(tmp)))
-    if timezone.now() - ndate > datetime.timedelta(7):
-        return date_format(ndate, "DATETIME_FORMAT")
-    return ndate.strftime("%a %H:%M")
+    ndate = datetime.datetime.fromtimestamp(email.utils.mktime_tz(tmp))
+    if ndate.tzinfo is not None:
+        tz = timezone.get_current_timezone()
+        ndate = tz.localize(datetime.datetime.fromtimestamp(ndate))
+    # No datetime formatting if USE_L10N is False
+    if not getattr(settings, "USE_L10N", False):
+        return ndate
+    if datetime.datetime.now() - ndate > datetime.timedelta(7):
+        return date_format(ndate, getattr(settings, "WEBMAIL_DATETIME_FORMAT", 'r'))
+    return date_format(ndate, getattr(settings, "WEBMAIL_SHORT_DATETIME_FORMAT", 'r'))
```

Comment:
I don't like the idea to use a setting for that...
Since modoboa allows users to customize the language, dates should be displayed according to that language. DATETIME_FORMAT is made for that (at least from what I understand when I read the documentation https://docs.djangoproject.com/en/1.9/ref/settings/#std:setting-DATETIME_FORMAT) and that's why I asked you to check if USE_L10n was enabled on your local installation...
Did I make a mistake somewhere ?